script/texdoclib-const.tlu

```diff
@@ -35,34 +35,215 @@ Try to find appropriate TeX documentation for the specified NAME(s).
 Alternatively, perform the given ACTION and exit.
 
 Options:
-  -w, --view        Use view mode: start a viewer. (default)
-  -m, --mixed       Use mixed mode (view or list).
-  -l, --list        Use list mode: show a list of results.
-  -s, --showall     Use showall mode: show also "bad" results.
-
-  -i, --interact    Use interactive menus. (default)
-  -I, --nointeract  Use plain lists, no interaction required.
-  -M, --machine     Machine-readable output for lists (implies -I).
-
-  -q, --quiet       Suppress warnings and most error messages.
-  -v, --verbose     Print additional information (e.g., viewer command).
-  -D, --debug       Activate all debug output (equal to "--debug=all").
-  -d LIST, --debug=LIST
-                    Activate debug output restricted to LIST.
-  -c NAME=VALUE     Set configuration item NAME to VALUE.
+{{mode}}
+
+{{interaction}}
+
+{{debug}}
 
 Actions:
-  -h, --help        Print this help message.
-  -V, --version     Print the version number.
-  -f, --files       Print the list of configuration files used.
-  --just-view FILE  Display FILE, given with full path (no searching).
+{{action}}
 
 Full manual available via `texdoc texdoc'.
 
 Website: <https://tug.org/texdoc/>
 Repository: <https://github.com/TeX-Live/texdoc>
 Please email bugs to <texdoc@tug.org>.]]
 
+zsh_completion = [[
+compdef __texdoc texdoc
+
+__texdoc() {
+  local options=(
+    {{action}}
+    + mode
+    {{mode}}
+    + interaction
+    {{interaction}}
+    + debug
+    {{debug}}
+  )
+  _arguments -C -A $options \
+    '*: :->arguments' && return
+  case $state in
+    arguments)
+      local tlpdb="$(kpsewhich -var-value TEXMFROOT)/tlpkg/texlive.tlpdb"
+      _values package $(awk '/^name[^.]*$/ {print $2}' $tlpdb)
+    ;;
+  esac
+}
+]]
+
+options = {
```

Comment:
Could you please add a multi-line comment to describe the data structure of the `options` table?

I noticed that you wrote the description in the commit message, but putting it as a comment in the source code is beneficial for future maintenance. Just like:

https://github.com/TeX-Live/texdoc/blob/b8f7ebd9eb14ad57d9d3f04e21fecaad349f6cf5/script/texdoclib-alias.tlu#L5-L21